engine/rest-client/src/main/java/org/datacleaner/restclient/ComponentList.java

```diff
@@ -50,11 +51,13 @@ public void add(ComponentInfo componentInfo) {
      */
     @JsonInclude(JsonInclude.Include.NON_NULL)
     public static class ComponentInfo {
-
         private String name = "";
         private String description = "";
         private String createURL = "";
+        private Set<String> categoryNames = new HashSet<>();
+        private String superCategoryName = null;
         private Map<String, PropertyInfo> properties = new HashMap<>();
+        private byte[] iconData = null;
```

Comment:
How will this serialize into JSON? Wondering if we should instead expose a URL for the icon.